pkg/test/testPackages.proj

```diff
@@ -176,19 +169,64 @@
     <Message Text="Generated $(_projectFile)" />
   </Target>
 
+  <Target Name="CopyProducedPackages">
+    <!-- Copy over package local cache to our helix runtime path -->
+    <ItemGroup>
+     <_localPackages Include="$(PackageOutputPath)*.nupkg" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(_localPackages)"
+          UseHardlinksIfPossible="true"
+          DestinationFolder="$(TestSupportDir)packages" />
+  </Target>
+
+  <Target Name="ArchiveHelixItems"
+          DependsOnTargets="GenerateProjects;CopyProducedPackages"
+          Condition="'$(ArchiveTests)' == 'true'">
+
+    <MakeDir Directories="$(TestArchiveTestsRoot)" />
+    <ZipDirectory
+        SourceDirectory="$(TestProjectDir)%(SupportedPackage.Identity)"
+        DestinationFile="$(TestArchiveTestsRoot)%(SupportedPackage.Identity).zip"
+        Overwrite="true" />
+
+    <MakeDir Directories="$(TestArchiveRuntimeRoot)" />
+    <ZipDirectory
+        SourceDirectory="$(TestSupportDir)"
+        DestinationFile="$(TestArchiveRuntimeRoot)packages-testPayload-$(ConfigurationGroup).zip"
+        Overwrite="true" />
+
+  </Target>
+
   <Target Name="RestoreProjects"
-          DependsOnTargets="GenerateProjects">
+          DependsOnTargets="GenerateProjects"
+          Condition="'$(ArchiveTests)' != 'true'">
     <Message Importance="High" Text="*** Restoring ***" />
     <Exec Command="$(TestRestoreCommand) &quot;$(TestProject)&quot;" EnvironmentVariables="@(CliEnvironment)" StandardOutputImportance="High" />
   </Target>
 
   <Target Name="BuildProjects"
-          DependsOnTargets="RestoreProjects">
-    <Message Importance="High" Text="*** Testing ***" />
-    <Exec Command="$(TestBuildCommand) &quot;$(TestProject)&quot; /p:TestPackages=%(SupportedPackage.Identity)" EnvironmentVariables="@(CliEnvironment)" StandardOutputImportance="High" />
+          DependsOnTargets="RestoreProjects"
+          Condition="'$(ArchiveTests)' != 'true'">
+    <Message Importance="High" Text="*** Testing *** %(SupportedPackage.Identity)" />
+    <Message Importance="High" Text="$(TestBuildCommand) &quot;$(TestProject)&quot;" />
+    <Exec Command="$(TestBuildCommand) &quot;$(TestProject)&quot;" EnvironmentVariables="@(CliEnvironment)" StandardOutputImportance="High" />
   </Target>
 
-  <Target Name="Build" DependsOnTargets="BuildProjects" />
+  <PropertyGroup>
+    <_BuildDependsOn>
+      BuildProjects
+    </_BuildDependsOn>
+  </PropertyGroup>
+
+  <PropertyGroup Condition="'$(ArchiveTests)' == 'true'">
```

Comment:
You already have this condition on the target itself, so its redundant here.  You can just append to the DependsOnTargets list.  Not worth defining a property since you don't expect folks to extend it.